### ISSUE TITLE
omitExtraData trims out required property groups

### ISSUE THREAD

**Issue by MatinF** [mozilla-services/react-jsonschema-form#1425]
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://mozilla-services.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

I've got a simple Schema example with two properties, each containing one property. The overarching properties are required to be in the formData, whereas the "sub properties" of each overarching property are optional. 

The problem is that omitExtraData seems to trim out the overarching properties by default.

### Steps to Reproduce

See playground link:
[Playground link](https://mozilla-services.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJnZW5lcmFsIjp7ImRldmljZSI6e30sInNlY3VyaXR5Ijp7fX19LCJzY2hlbWEiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsiZ2VuZXJhbCI6eyJ0aXRsZSI6IkdlbmVyYWwiLCJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJkZXZpY2UiOnsidGl0bGUiOiJEZXZpY2UiLCJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJtZXRhIjp7InRpdGxlIjoiTWV0YSBkYXRhIiwiZGVzY3JpcHRpb24iOiJYWVoiLCJ0eXBlIjoic3RyaW5nIiwibWluTGVuZ3RoIjowLCJtYXhMZW5ndGgiOjMwLCJwYXR0ZXJuIjoiXlthLXpBLVowLTk7IF0qJCJ9fX0sInNlY3VyaXR5Ijp7InRpdGxlIjoiU2VjdXJpdHkiLCJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJrcHViIjp7InRpdGxlIjoiU2VydmVyIHB1YmxpYyBrZXkiLCJkZXNjcmlwdGlvbiI6IkFCQyIsInR5cGUiOiJzdHJpbmciLCJtaW5MZW5ndGgiOjAsIm1heExlbmd0aCI6MTAwLCJwYXR0ZXJuIjoiXlthLXpBLVowLTkrLz1dKiQifX19fSwicmVxdWlyZWQiOlsiZGV2aWNlIiwic2VjdXJpdHkiXX19fSwidWlTY2hlbWEiOnt9LCJsaXZlU2V0dGluZ3MiOnsidmFsaWRhdGUiOnRydWUsImRpc2FibGUiOmZhbHNlLCJvbWl0RXh0cmFEYXRhIjp0cnVlLCJsaXZlT21pdCI6dHJ1ZX19)

When clearing the formData, the default formData is created - which looks as expected. Further, entering something in one of the property fields and then deleting this content leads to the default formData again.

#### Expected behavior

I would expect that when I hit submit, I get the output equal to the formData - i.e. below:

```
{
  "general": {
    "device": {},
    "security": {}
  }
}
```

This output should come regardless of whether I simply load the Schema and immediately hit submit - or whether I enter something in a field and then deletes that again.

#### Actual behavior

Instead, if I simply load the Schema and hit submit with the default formData (without editing the fields), none of the properties are included in the output (see the console). Instead I get an empty formData output in the console.

If I change one of the property fields in the editor and clear that data again in the editor, the formData looks the same as the default - but now the console "correctly" outputs the property that I edited (but not the other property until I also edit this).

This issue is rather critical for cases where users want to ensure that the formData contains the expected "skeleton" structure (in our case the `device` and `security` fields) when these are included in the Schema, at least when they're explicitly required.

I would expect the omitExtraData to primarily be used for eliminating property fields that are not included in the Schema - not the properties actually listed in the Schema.

### Version

This was done August 22, using the playground.

**Comment by Tonexus** [comment]
FYI, this is fixed in #1419.

**Comment by MatinF** [comment]
Thanks, looks to be resolved
